bridge-adapters/kinetic-bridgehub-adapter-cherwell/src/main/java/com/kineticdata/bridgehub/adapter/cherwell/CherwellAdapter.java

```diff
@@ -0,0 +1,758 @@
+        
+        // Path builder functions may mutate the parameters Map;
+        String path = mapping.getPathbuilder().apply(structureList, parameters);
+                
+        // Accessor values is either passed as a parameter in the qualification
+        // mapping for Adhoc or on the mapping for all other structures.
+        String accessor = getAccessor(mapping, parameters);
+        
+        
+        Map<String, NameValuePair> parameterMap = buildNameValuePairMap(parameters);
+
+        // Retrieve the objects based on the structure from the source
+        JSONObject responseObject = apiHelper.executeGetRequest(getUrl(path, parameterMap));
+        
+        JSONArray responseArray;
+        if (responseObject.containsKey(accessor)) {
+            responseArray = getResponseData(responseObject.get(accessor));
+        } else {
+            responseArray = getResponseData(responseObject);
+        }
+        
+        Record record = new Record();
+        if (responseArray.size() == 1) {
+            // Reassign object to single result 
+            JSONObject object = (JSONObject)responseArray.get(0);
+                
+            List<String> fields = getFields(request.getFields() == null ? 
+                new ArrayList() : request.getFields(), object);
+            record = buildRecord(fields, object);
+        } else if (responseArray.isEmpty()) {
+            LOGGER.debug("No results found for query: {}", request.getQuery());
+        } else {
+            throw new BridgeError ("Retrieve must return a single result."
+                + " Multiple results found.");
+        }
+
+        // Return the created Record object
+        return record;
+    }
+
+    @Override
+    public RecordList search(BridgeRequest request) throws BridgeError {
+        // Log the access
+        LOGGER.trace("Searching Records");
+        LOGGER.trace("  Structure: " + request.getStructure());
+        if (request.getQuery() != null) {
+            LOGGER.trace("  Query: " + request.getQuery());
+        }
+        if (request.getFieldString() != null) {
+            LOGGER.trace("  Fields: " + request.getFieldString());
+        }
+        
+         // parse Structure
+        List<String> structureList = Arrays.asList(request.getStructure().trim()
+            .split("\\s*>\\s*"));
+        // get Structure model
+        AdapterMapping mapping = getMapping(structureList.get(0));
+        
+        Map<String, String> parameters = getParameters(
+            parser.parse(request.getQuery(),request.getParameters()), structureList);
+        
+        Map<String, String> metadata = request.getMetadata() != null ?
+                request.getMetadata() : new HashMap<>();
+
+        // If form defines sort order use it. This will overwrite $orderby in the
+        // qualification mapping.
+//        if (metadata.get("order") != null) {
+//            parameters.put("OrderBy", addSort(metadata.get("order")));
+//        }
```

Comment:
Was this forgotten? Should it be removed?

Reply:
Ya, I've asked Matt H how ordering works with Cherwell but he has not gotten back to me.  I'm assumeing that it will need to be addressed later.  I'll remove the code for now.